### DISCUSSION TITLE
Alternative mode for small displays: show tree as temporary popover, instead of sidebar

### DISCUSSION THREAD

**Discussion by ELLIOTTCABLE** [piroor/treestyletab#3255] [Ideas]
I absolutely _adore_ Tree Style Tab. It is, genuinely, the reason I use FireFox — I'm going to use whichever browser has this addon, not use whichever addon is available for my preferred browser.

One of the only constant complaints, really, is how space-inefficient it is. On a small-ish screen, fully **⅙ of my entire screen** is wasted, mostly, on empty, grey, negative space — especially before I've really gotten into my tasks of the day, and there's not many tabs open; or on "secondary" browser-windows where I'm not intending to open many tabs:

<img width="512" alt="Screen Shot 2022-11-25 at 3 46 52 PM (2)" src="https://user-images.githubusercontent.com/200/204058148-2c7f0165-d20e-45d1-b4fe-dfdeba968941.png">

I'd love an alternative mode (_especially_ if it can be activated quickly and easily) that hides the sidebar, just like clicking the addon's icon currently does … but results in _future_ clicks of the icon dropping down a _temporary popover_ version of the tree; especially and importantly, one that only takes up as much screenspace as there are tabs: (forgive the absolutely terrible mockup, lol)

<img width="661" alt="Screen Shot 2022-11-25 at 3 51 03 PM (2)" src="https://user-images.githubusercontent.com/200/204058443-1f0ee89c-0f85-4b81-baee-6024bb16f7b4.png">

**Comment by irvinm** [comment]
Another thing that you can do today is use Firefox CSS to auto-expand\contract the sidebar when hovering.  (Reddit r/FirefoxCSS is probably the best place to check)  Obviously @piroor will chime in whether he wants or can implement what you are asking for from the button.

The challenge with the hover approach is the upkeep when CSS elements change over time and formatting the behavior for what you are looking for specifically.  However, it seems there are enough people using this type of solution, that there should be help on Reddit.

**Comment by piroor** [comment]
Another solution, there is a helper addon to open TST's sidebar in a separate window: [Tree Style Tab in Separate Window](https://addons.mozilla.org/firefox/addon/tst-in-separate-window/) made by @Lej77 .

Currently TST's sidebar panel requires initialization on each showing and it takes time - indeed you have to wait for a while when you toggle TST's sidebar panel. Firefox's panel view always unloads and reloads its contents so it will be stressful. For a better performance a permanently opened floating window for TST's panel looks a realistic way.

There is a concern: the size of the floating window. Currently the addon TST in Separate Window does not provide ability to resize its window by hovering. I think it can be implemented with some APIs to listen mouse events on the panel: https://github.com/piroor/treestyletab/wiki/API-for-other-addons#when-the-pointer-is-moved-over-tabs
